cores/esp32/esp32-hal-uart.h

```diff
@@ -68,17 +68,12 @@ void uartFlushTxOnly(uart_t* uart, bool txOnly );
 void uartSetBaudRate(uart_t* uart, uint32_t baud_rate);
 uint32_t uartGetBaudRate(uart_t* uart);
 
-size_t uartResizeRxBuffer(uart_t* uart, size_t new_size);
-
 void uartSetRxInvert(uart_t* uart, bool invert);
 
 void uartSetDebug(uart_t* uart);
 int uartGetDebug();
 
-void uartStartDetectBaudrate(uart_t *uart);
-unsigned long uartDetectBaudrate(uart_t *uart);
```

Comment:
so are we losing the baud rate detection for all chips?

Reply:
As I mentioned before this functionality only works for ESP32-C3.
It doesn't work for ESP32 neither for ESP32-S2.
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/esp32-hal-uart.c#L720-L775

It is used in HardwareSerial::begin(...)
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/HardwareSerial.cpp#L88-L107

It allows using Serial.begin(0);
But for ESP32 and ESP32-S2 doing it will cause a problem... given that it doesn't do anything.

Thus to do not create false expectation to 99% of Arduino users, I decided to remove it from the code.
It can be placed as some Library for the C3 only if necessary. 

It will clearer for the end user.

Summary:
Leave it out as proposed in the PR.

Reply:
Actually it only does NOT work on C3 :) look at the `#ifndef`

Reply:
oops.... got it. 
I think it would be better to change it to #ifdef instead, thinking in the future roadmap of Arudino supported SoCs.
Or C3 will be the only one to do not support baudrateDetection?

Anyway I'll add it back!

Reply:
it has slightly different registers probably and since I could not test it at the time, I excluded it :) should be implementable now

Reply:
https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32c3/include/soc/esp32c3/include/soc/uart_struct.h#L208-L228

https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32c3/include/soc/esp32c3/include/soc/uart_struct.h#L190

Test this if you can:
```cpp

/*
 * To start detection of baud rate with the uart the auto_baud.en bit needs to be cleared and set. The bit period is 
 * detected calling uartBadrateDetect(). The raw baudrate is computed using the UART_CLK_FREQ. The raw baudrate is 
 * rounded to the closed real baudrate.
*/
void uartStartDetectBaudrate(uart_t *uart) {
  if(!uart) return;
#ifdef CONFIG_IDF_TARGET_ESP32C3
  uart->dev->rx_filt.glitch_filt = 0x08;
  uart->dev->rx_filt.glitch_filt_en = 1;
  uart->dev->conf0.autobaud_en = 0;
  uart->dev->conf0.autobaud_en = 1;
#else
  uart->dev->auto_baud.glitch_filt = 0x08;
  uart->dev->auto_baud.en = 0;
  uart->dev->auto_baud.en = 1;
#endif
}

unsigned long
uartDetectBaudrate(uart_t *uart)
{
    static bool uartStateDetectingBaudrate = false;

    if(!uartStateDetectingBaudrate) {
        uartStartDetectBaudrate(uart);
        uartStateDetectingBaudrate = true;
    }

    unsigned long divisor = uartBaudrateDetect(uart, true);
    if (!divisor) {
        return 0;
    }

#ifdef CONFIG_IDF_TARGET_ESP32C3
    uart->dev->conf0.autobaud_en = 0;
#else
    uart->dev->auto_baud.en = 0;
#endif
    uartStateDetectingBaudrate = false; // Initialize for the next round

    unsigned long baudrate = getApbFrequency() / divisor;

    static const unsigned long default_rates[] = {300, 600, 1200, 2400, 4800, 9600, 19200, 38400, 57600, 74880, 115200, 230400, 256000, 460800, 921600, 1843200, 3686400};

    size_t i;
    for (i = 1; i < sizeof(default_rates) / sizeof(default_rates[0]) - 1; i++)	// find the nearest real baudrate
    {
        if (baudrate <= default_rates[i])
        {
            if (baudrate - default_rates[i - 1] < default_rates[i] - baudrate) {
                i--;
            }
            break;
        }
    }

    return default_rates[i];
}
```

Reply:
Unfortunatelly C3 didn't work correctly. It reports wrong baudrate.

Reply:
I haven't been able to compile these for testing. 
I'm getting file not found errors. E.G. C:\Users\Ralph\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.6\cores\esp32\esp32-hal-uart.c:27:25: fatal error: hal/uart_ll.h: No such file or directory
compilation terminated

Reply:
@RLmonitor,
In order to use this PR, it's necessary to use Arduino Core 2.0.0RC version. I see that you are using 1.0.6.

When using Arduino IDE, it is a change on the Board Manager to update to version 2.0.0RC.

For PlatformIO, it will be necessary to wait for them to update their servers, which only will happen when we realease officially the 2.0.0 version.

Core 1.0.6 uses IDF version 3.3 which is incompatible with IDF 4.4 used in Arduino Core 2.0.0.